CHANGES.rst

```diff
@@ -122,6 +122,11 @@ API Changes
   - Angstrom, erg, G, and barn are no more reported as deprecated FITS units.
     [#5929]
 
+  - Remove compatibility code which forced loading all HDUs on close. The old
+    behavior can be used with ``lazy_load_hdus=False``. Because of this change,
+    trying to access the ``.data`` attribute from an HDU which is not loaded now
+    raises a IndexError instead of a ValueError. [#6082]
```

Comment:
Ok, I found the two versions in the CHANGES file, so I was not sure which one is preferred.